crates/composefs-boot/src/bootloader.rs

```diff
@@ -144,7 +145,7 @@ impl<ObjectID: FsVerityHashValue> Type1Entry<ObjectID> {
     // This is a bit of a strange operation: for each file mentioned in the bootloader entry, move
     // the file into the given 'entry_id' pathname and rename the entry file itself to
     // "{entry_id}.conf".
-    pub fn relocate(&mut self, entry_id: &str) {
+    pub fn relocate(&mut self, entry_id: &str, boot_subdir: Option<&str>) -> Result<()> {
```

Comment:
Same nags as before: try to keep these in the order that they get pasted back together... so when they appear, they are always in the order of: boot_partition, boot_subdir, entry_id.

Reply:
Also: with the arguments defined as they are now, I don't think this function needs to be failable anymore...